### ISSUE TITLE
Riot don't ring

### ISSUE THREAD

**Issue by B4rabbas** [vector-im/riot-android#2759]
Riot don't work if the phone is locked

**Comment by Buntbart** [comment]
It was the same here until I switched off the battery optimization for Riot.im.

**Comment by BillCarsonFr** [comment]
Hi @B4rabbas, can you please submit a bug report by shacking your device (or using the Report Bug item in the app menu). It will allow me to investigate the logs.

**Comment by B4rabbas** [comment]
@BillCarsonFr done, I've mentionned your name in the bug report

**Comment by BillCarsonFr** [comment]
Thanks @B4rabbas is saw it.
I can see that you are using the FDroid version. Just to be sure, when you say that it doesnt ring does that mean that you have no notifications or that the notification does not make sound?
In the FDroid version, you should always have a 'sticky' notification in the system tray (saying listening for events) is it the case?
Thx.

**Comment by B4rabbas** [comment]
Yes there is always the sticky notification of riot (and battery optimisation disabled too).
There is no sound and no ring neither.

**Comment by B4rabbas** [comment]
I see the notification only if I open the app, with missed vocal or video call.

**Comment by B4rabbas** [comment]
I have seen the same issue on 5 phones, xiaomi redmi 5 plus, huawei honor 7, asus z2, huawei mate 20 pro & samsung j6 2016. With fdroid version at least.

**Comment by bennypr0fane** [comment]
I get no audio for notifications with the Google Play version on Sony Xperia X (F5121)

**Comment by B4rabbas** [comment]
If only Riot would work it definitly destroy wathsapp & co...

**Comment by szimszon** [comment]
If I understand it well in battery saving mode Riot don't fetch the message from the server until in foreground. So if someone calls you only a push notif. is sent through Google and Android takes it as a notification (with a notification sound).
In normal mode Riot receives the push notification through Google and contacts the server, sees it is a call and acts accordingly. Developers may correct me if I'm wrong here.

**Comment by B4rabbas** [comment]
Any news ?

**Comment by bmarty** [comment]
Sorry we have been quite busy with the keys backup feature.
Notification issues will be our P1 subject for February

**Comment by B4rabbas** [comment]
So nice, be sure I'm  following this

**Comment by gerroon** [comment]
I have this issue with F-droid version where the ringing and call notification does not happen whatsoever as long as the app is not in the foreground. 

So the app must be open and be foreground app to take calls, otherwise I only get "missed call" or "dropped call" notification about a cali that was never engaged on my device. 

Google Play version does not seem to have this issue.

**Comment by sylph1o** [comment]
> I have this issue with F-droid version where the ringing and call notification does not happen whatsoever as long as the app is not in the foreground.
> 
> So the app must be open and be foreground app to take calls, otherwise I only get "missed call" or "dropped call" notification about a cali that was never engaged on my device.

Same here. No notification at all, not even for a missed call (F-Droid version too).

If the app is opened while being called, the call notification happens as expected, with the screen to answer (or not) the call.

If the app is opened _after missing the call_, the ringtone will play and the 'incoming call' notification will be received (the one that cannot be dismissed). The screen on which to answer (or not) the call will not appear, though.

I cannot report on the Google Play version. In my humble and respectful opinion, not being able to receive a call is quite a serious problem when trying to convince people to switch over from (in)famous proprietary alternatives.

**Comment by bennypr0fane** [comment]
In your reports, please be sure to differentiate between
### "no notification" (and hence no audio signal), which is not the topic of this issue - there's another one about that problem - and
### notification is received, but not accompanied by audio signal, which is the topic of this issue.

**Comment by bennypr0fane** [comment]
I haven't seen a single Instant Messaging app for Android that has notifications working correctly without using FCM.
I am hence beginning to suspect that this is just technically impossible, and it's by design. It may be a feature of Android that Messaging services can't work properly unless they use FCM

**Comment by sylph1o** [comment]
Thank you @bennypr0fane, I reported in the wrong issue, then. Do you have the reference of the issue about the absence of call notification when Riot is not on the foreground?

**Comment by gerroon** [comment]
> I haven't seen a single Instant Messaging app for Android that has notifications working correctly without using FCM (I've tried more than half a dozen) .
> I am hence beginning to suspect that this is just technically impossible, and it's by design. It may be a feature of Android that Messaging services can't work properly unless they use FCM

Notifications work with the Fdroid version it is the ringing that does not work

**Comment by bennypr0fane** [comment]
> Thank you @bennypr0fane, I reported in the wrong issue, then. Do you have the reference of the issue about the absence of call notification when Riot is not on the foreground?
It's here: [#2800](https://github.com/vector-im/riot-android/issues/2800)
@gerroon : It's good to know that notifications are working for you in the F-Droid version, but alas, not for everyone. Maybe it can be useful for you to report you device model/Android version working in raltion to that issue.

**Comment by GammaPi** [comment]
Huawei EMUI is famous for killing apps ruthlessly.
The solutions on dontkillmyapp.com works and the steps include:
1.Use adb to uninsall powerengine
2.Lock the app in recent app page (and don't swipe up if you want to receive calls when the screen is locked)
3.Enable all 3 switches in app launch setting
4.Battery optimization
5.Open "Optimizer/Tablet Manager" app and disable Smart tune-up (or huawei will reverse all the settings automatically after sometime)
